File: cffi_dist/example_csharp/Cookies.cs
namespace TlsClientExamples;

internal static class Cookies
{
    private const string SessionId = "my-cookie-session";

    internal static void Run()
    {
        // The server sets a cookie on this request; the session's cookie jar
        // stores it automatically and replays it on every later request in
        // this session.
        var response = TlsClientLibrary.Request(new
        {
            tlsClientIdentifier = "chrome_150",
            followRedirects = true,
            sessionId = SessionId,
            proxyUrl = "",
            headers = new Dictionary<string, string> { ["accept"] = "*/*" },
            headerOrder = new[] { "accept" },
            requestUrl = "https://httpbin.org/cookies/set?session=abc123",
            requestMethod = "GET",
            requestBody = "",
            requestCookies = Array.Empty<object>()
        });

        TlsClientLibrary.FreeMemory((string)response["id"]);

        var cookies = TlsClientLibrary.GetCookiesFromSession(new
        {
            sessionId = SessionId,
            url = "https://httpbin.org"
        });

        Console.WriteLine(cookies);

        TlsClientLibrary.DestroySession(new { sessionId = SessionId });
    }
}

File: cffi_dist/example_typescript/src/samples/cookies.ts
import {TLSClientFetchCookiesForSessionRequestPayload, TLSClientRequestPayload} from "@project/types";
import {TLSClient} from "@project/client";

const tlsClient = new TLSClient()

const sessionId = 'my-cookie-session';

// The server sets a cookie on this request; the session's cookie jar stores
// it automatically and replays it on every later request in this session.
const payload: TLSClientRequestPayload = {
    tlsClientIdentifier: 'chrome_150',
    followRedirects: true,
    sessionId: sessionId,
    proxyUrl: '',
    headers: {'accept': '*/*'},
    headerOrder: ['accept'],
    requestUrl: 'https://httpbin.org/cookies/set?session=abc123',
    requestMethod: 'GET',
    requestBody: '',
    requestCookies: []
};

tlsClient.request(payload);

const fetchCookiesPayload: TLSClientFetchCookiesForSessionRequestPayload = {
    sessionId: sessionId,
    url: 'https://httpbin.org',
};

const cookiesInSession = tlsClient.getCookiesFromSession(fetchCookiesPayload);

console.log(cookiesInSession);

tlsClient.destroySession({sessionId: sessionId});
